src/pages/RFOX/components/Claim/Claim.tsx

```diff
@@ -0,0 +1,99 @@
+import { AnimatePresence } from 'framer-motion'
+import { lazy, Suspense, useCallback, useState } from 'react'
+import { MemoryRouter, Route, Switch, useLocation } from 'react-router'
+import { makeSuspenseful } from 'utils/makeSuspenseful'
+
+import type { RfoxClaimQuote } from './types'
+import { ClaimRoutePaths, type ClaimRouteProps } from './types'
+
+const suspenseFallback = <div>Loading...</div>
+
+const ClaimSelect = makeSuspenseful(
+  lazy(() =>
+    import('./ClaimSelect').then(({ ClaimSelect }) => ({
+      default: ClaimSelect,
+    })),
+  ),
+)
+
+const ClaimConfirm = makeSuspenseful(
+  lazy(() =>
+    import('./ClaimConfirm').then(({ ClaimConfirm }) => ({
+      default: ClaimConfirm,
+    })),
+  ),
+)
+
+const ClaimStatus = makeSuspenseful(
+  lazy(() =>
+    import('./ClaimStatus').then(({ ClaimStatus }) => ({
+      default: ClaimStatus,
+    })),
+  ),
+)
+
+const ClaimEntries = [ClaimRoutePaths.Select, ClaimRoutePaths.Confirm, ClaimRoutePaths.Status]
+
+export const Claim: React.FC<ClaimRouteProps> = ({ headerComponent }) => {
+  return (
+    <MemoryRouter initialEntries={ClaimEntries} initialIndex={0}>
+      <ClaimRoutes headerComponent={headerComponent} />
+    </MemoryRouter>
+  )
+}
+
+export const ClaimRoutes: React.FC<ClaimRouteProps> = ({ headerComponent }) => {
+  const location = useLocation()
+
+  const [claimQuote, setClaimQuote] = useState<RfoxClaimQuote | undefined>()
+  const [claimTxid, setClaimTxid] = useState<string | undefined>()
+
+  const renderClaimSelect = useCallback(() => {
+    return <ClaimSelect headerComponent={headerComponent} setClaimQuote={setClaimQuote} />
+  }, [headerComponent])
+
+  const renderClaimConfirm = useCallback(() => {
+    if (!claimQuote) return null
+
+    return (
+      <ClaimConfirm
+        claimQuote={claimQuote}
+        setClaimTxid={setClaimTxid}
+        headerComponent={headerComponent}
+      />
+    )
+  }, [claimQuote, headerComponent])
+
+  const renderClaimStatus = useCallback(() => {
+    if (!claimTxid) return null
+    if (!claimQuote) return null
```

Comment:
tyvm for the initial API honouring 💜 will make for less boilerplate assuming we do things the exact same way as other tabs (which we most likely will)